index.js

```diff
@@ -244,6 +246,83 @@ class GithubScm extends Scm {
             }
         };
     }
+
+    /**
+     * Get data for a specific repo
+     * @method getCommitSha
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.scmUrl     The scmUrl to get data of
+     * @param  {String}   config.token      The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getRepoInfo(scmInfo, token) {
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand({
+                action: 'get',
+                token,
+                params: scmInfo
+            }, (error, repoInfo) => {
+                if (error) {
+                    return reject(error);
+                }
+
+                return resolve(repoInfo);
+            });
+        });
+    }
+
+    /**
+     * Get a url for a specific branch
+     * @method getRepoId
+     * @param  {Object}    config        Configuration
+     * @param  {String}    config.scmUrl The scmUrl to get url of
+     * @param  {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getBranchUrl(scmInfo, token) {
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand({
+                action: 'getBranch',
+                token,
+                params: scmInfo
+            }, (error, branchInfo) => {
+                if (error) {
+                    return reject(error);
+                }
+
+                // eslint-disable-next-line no-underscore-dangle
+                return resolve(branchInfo._links.html);
+            });
+        });
+    }
+
+    /**
+     * Get repository object for storing new pipeline
+     * @method getRepoId
+     * @param {Object}    config        Configuration
+     * @param {String}    config.scmUrl The scmUrl to generate ID
+     * @param {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getRepoId(config) {
```

Comment:
This is a little misleading I think. I would expect it to return only the `id`, but it actually returns an object with other things, and I need to do `result.id` to get it. 

Now that I reread the scm-base, it says that Returns a unique id
https://github.com/screwdriver-cd/scm-base/blob/master/index.js#L133
but it actually returns an object. 

I think we should either change this function name to be less misleading, or just returns the id (if we don't use name and url)